packages/dev/sharedUiComponents/src/fluent/hoc/propertyLines/vectorPropertyLine.tsx

```diff
@@ -37,13 +36,16 @@ export type TensorPropertyLineProps<V extends Vector3 | Vector4 | Quaternion> =
         };
     };
 
+const HasZ = (vector: Vector2 | Vector3 | Vector4 | Quaternion): vector is Vector3 => !(vector instanceof Vector2);

```

Comment:
I'm wondering if we should be making these all work with the `IVector3Like` type interfaces, since we use those in some parts of the API, and if any of those use cases need to be bound to UI, it won't work if these components only work with the concrete classes.